enhancedautoclicker.js

```diff
@@ -270,7 +270,12 @@ function autoGym() {
 
             if (App.game.gameState != GameConstants.GameState.gym) {
                 //If "All" is selected, then go through list of league from 0-4
-                if(gymSelect === 5) {
+                if(gymSelect === 5 && (((player.town().content[0] instanceof Gym && player.town().content[0].isUnlocked()) &&
```

Comment:
You don't need to check if they are all unlocked, you can just check 

`if (gymSelect === 5 && (player.town().content[gymSelect] instanceof Gym && player.town().content[gymSelect].isUnlocked()) || (player.town().content[gymSelect] instanceof Champion && player.town().content[gymSelect].isUnlocked()))`

Reply:
Updated my patch to include this instead